src/librustc_mir/transform/const_prop.rs

```diff
@@ -525,23 +525,20 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
 
                 Ok(())
             })?;
-        } else if let Rvalue::Ref(_, _, place) = rvalue {
+        } else if let Rvalue::Ref(_, _, Place {
+                                            base: PlaceBase::Local(local),
```

Comment:
the indentation is a bit off here

Reply:
maybe this if let else chain should be a match?

Reply:
Fixed (I think)

Reply:
It's not supposed to be exhaustive, just checking for a few specific cases. I'm not sure what the convention is in Rust for that. I don't mind changing it though...

Reply:
Having a `_ => {}` arm is totally fine. I personally prefer matching if all if conditions pattern match the same value or compare the same value with constants. It makes it clearer that all if/else components make their decision on a single value.

Reply:
Fixed